### PR TITLE
fix: update validate and evaluate to support target flag

### LINKED ISSUE #18: Update Validate Action to Support Target
### Describe what should be investigated or refactored

With the v0.4.5 of Lula, a new flag `-t` has been created that allows for targeting a specific framework prop applied to the `implemented-requirements` in the component definition.

### Links to any relevant code

https://github.com/defenseunicorns/lula/blob/main/docs/ns/target.md

### LINKED ISSUE #23: Create a test workflow
### Describe what should be investigated or refactored

Creation of a test workflow that lives in this repository and tests the actions on pull request. Ideally performs:
- version validation (IE the version of lula during setup is what was specified or default)
- different optionality for validate and evaluate

### Additional context

May require some mock oscal to live in this repository - does not require any lula validations. Lula has a multi-component oscal file which has multiple components with multiple control-implementations which may help ferret out specific options.

### PR DESCRIPTION
**Pull request by CloudBeard**
## Description

Update the `validate` and `evaluate` action to check if `target` is empty or not. If `target` is empty processes as normal. If `target` exists will use the framework provided to run the `validate` and `evaluate` commands against that framework.

Changed the variable wording. The old target was the file `input-file`, with the addition of the `target` flag that could be confusing reading the code so refactored the variables to allow for `target` to represent the lula `target` flag and `oscal-file` to represent the `input-file`

removed the `-a` flags from the lula commands and replaced with `-o` to output the files correctly.

## Related Issue

Fixes #18 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
Updated Documentation to include the new flag. 

Tested the bash script to ensure functions correctly.

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/lula-action/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Review by brandtkeller — CHANGES_REQUESTED** [review]
I believe intent is correct - I'd like to get ahead of a few more configurations:

1. The output of validate should be configurable. This allows CI/Lula to merge assessments natively as part of validate output. 
2. I'd like to consider if we should split validate / evaluate or not. I don't hold a strong opinion at the moment but integration work we've done to-date may help influence them being done separately.

**Comment by CloudBeard** [comment]
Split those into separate actions, updated docs, and added an options field to future proof it a bit.

File: validate/action.yml [hunk 1]
```diff
@@ -1,41 +1,48 @@
...
-        #!/bin/bash
+       #!/bin/bash
         # Base url for lula artifacts
-        target_dir="${HOME}/.lula-cache"
-        threshold="${{ inputs.threshold }}"
-        target="${{ inputs.oscal-target }}"
+        file_dir="${HOME}/.lula-cache"
+        assessment-result="${{ inputs.assessment-result }}"
+        component-definition="${{ inputs.component-definition }}"
+        target="${{ inputs.target }}"
```
Comment (brandtkeller) [inline validate/action.yml:24]: Calling out for discussion - is `target` something that the action needs to natively support if `options` exists? 

or rather _should_ it? I don't feel strongly either way but I am looking for how to reduce the amount of conditionals present in both actions.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Appears valid but I'd either need to instrument a test repo or run locally. As such I am going to create an issue to create a workflow for testing the actions that can run in these pipelines. 

I tossed a few comments out for discussion - push back as you see fit but i'd like to reduce complexity as much as possible.

**Comment by CloudBeard** [comment]
Removed target and added multiple tests in the workflow to cover a few possible options.

Should also close https://github.com/defenseunicorns/lula-action/issues/23

**Review by brandtkeller — APPROVED** [review]
Appreciate the iterations that lead to cleaning these up.

**Pull request merged**